### PR TITLE
[modular] longarm jumpscare: mild CMG damage buff in return for making the stock actually collapse

### PR DESCRIPTION
**Pull request by Hatterhat**
## About The Pull Request
CMG buff:
- damage multiplier nudged from 0.5x to 0.7x, per-shot raw damage calculated in comments

CMG nerfs:
- base size increased to bulky
- can have its stock collapsed with altclick + 1 second of standing still + holding it to return it to normal size
- collapsed-stock CMGs have 20 spread, up from 0
- extended-stock CMGs have 0 spread

fun bug: did you know that autofire ignores a gun's safety
## How This Contributes To The Skyrat Roleplay Experience
making the CMG slightly better for shooting people with. unfortunately the WT-550 now continues to leave something to be desired

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
 
image
image

</details>

## Changelog

:cl:
balance: Thanks to further developments from Nanotrasen Arms' R&D Division, the CMG-2's rounds now carry slightly more stopping power than before (0.5x -> 0.7x damage multiplier).
balance: Unfortunately, adjustments to the CMG's internals now make its folding stock actually adjustable, and also matter for use. Ctrl-click to toggle its' stock. Users are warned that improper handling will lead to decreased accuracy.
/:cl:

### CODE REVIEW

**Comment by kittysmooch** [comment]
does extendo stock fit on suit slots?

File: modular_skyrat/modules/sec_haul/code/guns/cmg.dm [hunk 1]
```diff
@@ -21,7 +25,65 @@
...
+	/// how long does it take to extend/collapse the stock
+	var/toggle_time = 1 SECONDS
+	/// what's our spread with our extended stock (mild varedit compatibility I Guess)?
+	var/unfolded_spread = 0
+	/// what's our spread with a folded stock (see above comment)?
+	var/folded_spread = 20
+
+/obj/item/gun/ballistic/automatic/cmg/examine(mob/user)
+	. = ..()
+	. += span_notice("<b>Alt-click</b> to [folded ? "extend" : "collapse"] the stock. This takes a moment.")
```
Comment (tf-4) [inline modular_skyrat/modules/sec_haul/code/guns/cmg.dm]:
```suggestion
	. += span_notice("<b>Alt-click</b> to [folded ? "extend" : "collapse"] the stock.")
```
How does the alt click interaction play with silencers?

File: modular_skyrat/modules/sec_haul/code/guns/cmg.dm [hunk 2]
```diff
@@ -21,7 +25,65 @@
...
+	. = ..()
+
+/obj/item/gun/ballistic/automatic/cmg/proc/toggle_stock(mob/user, var/forced)
+	if(!user && forced) // for the possible case of having every shipped CMG be pre-folded
+		folded = !folded
+		update_fold_stats()
+		return
+	balloon_alert(user, "[folded ? "extending" : "collapsing"] stock...")
+	if(!do_after(user, toggle_time))
+		balloon_alert(user, "failed!")
```
Comment (tf-4) [inline modular_skyrat/modules/sec_haul/code/guns/cmg.dm]:
```suggestion
		balloon_alert(user, "interrupted!")
```
Just a consistency thing

File: modular_skyrat/modules/sec_haul/code/guns/cmg.dm [hunk 3]
```diff
@@ -21,7 +25,65 @@
...
+		return
+	folded = !folded
+	update_fold_stats()
+	balloon_alert(user, "stock [folded ? "collapsed" : "extended"]!")
```
Comment (tf-4) [inline modular_skyrat/modules/sec_haul/code/guns/cmg.dm]:
```suggestion
	balloon_alert(user, "stock [folded ? "collapsed" : "extended"]")
```
`!`s are for fail states

**Review by tf-4 — CHANGES_REQUESTED** [review]

**Review by Gandalf2k15 — APPROVED** [review]
Ideally this should be a component but out of scope. Someday maybe it will be.

**Pull request merged**